File: archive/g/go/prime-number.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func isPrime(n int) bool {
	if n < 2 {
		return false
	} else {
		for i := 2; i <= n/2; i++ {
			if n%i == 0 {
				return false
			}
		}
	}
	return true
}

func exitWithError() {
	fmt.Println("Usage: please input a non-negative integer")
	os.Exit(1)
}

func main() {
	if len(os.Args) != 2 {
		exitWithError()
	}

	n, err := strconv.Atoi(os.Args[1])
	if err != nil || n < 0 {
		exitWithError()
	}

	if isPrime(n) {
		fmt.Println("Prime")
	} else {
	    fmt.Println("Composite")
	}
}

File: archive/s/swift/prime-number.swift
import Foundation

let usage = "Usage: please input a non-negative integer"

if CommandLine.arguments.count != 2 {
    print(usage)
    exit(0)
}

guard let number = Int(CommandLine.arguments[1]), number >= 0 else {
    print(usage)
    exit(0)
}

if number < 2 {
    print("composite")
    exit(0)
}

var isPrime = true
let limit = Int(Double(number).squareRoot())

if limit >= 2 {
    for i in 2...limit {
        if number % i == 0 {
            isPrime = false
            break
        }
    }
}

print(isPrime ? "prime" : "composite")